common/datavalues/src/columns/primitive/mod.rs

```diff
@@ -189,17 +189,32 @@ impl<T: PrimitiveType> Column for PrimitiveColumn<T> {
         if length == self.len() {
             return Arc::new(self.clone());
         }
-        let iter = self
-            .values()
-            .iter()
-            .zip(filter.values().iter())
-            .filter(|(_, f)| *f)
-            .map(|(v, _)| *v);
 
-        let values: Vec<T> = iter.collect();
-        let col = PrimitiveColumn {
-            values: values.into(),
-        };
+        let mut res = Vec::<T>::with_capacity(length);
+        let mut offset = 0;
+        let values = self.values();
+
+        const MASK_BITS: usize = 64;
+        for mut mask in filter.values().chunks::<u64>() {
```

Comment:
should care about the remaining in chunks.

Reply:
I  have checked that `Bitmap::chunks()` won't yield chunk less than 64

Reply:
In my unittest, I set N = 1000 to make Column not aligned to 64.